src/settings/StaffSlips/StaffSlipDetail.test.js

```diff
@@ -95,71 +85,118 @@ describe('StaffSlipDetail', () => {
-        }), {});
-      });
+    it('should render "Template content" label', () => {
+      expect(screen.getByText(labelIds.staffSlipsTemplateContent)).toBeDefined();
+    });
 
-      it('should open and close PreviewModal component', () => {
-        PreviewModal.mockImplementationOnce(({
-          onClose,
-          'data-testid': testId,
-        }) => (
-          // eslint-disable-next-line jsx-a11y/click-events-have-key-events, jsx-a11y/no-static-element-interactions
-          <div
-            data-testid={testId}
-            onClick={onClose}
-          />
-        ));
-
-        expect(PreviewModal).toHaveBeenLastCalledWith(expect.objectContaining({
-          open: false,
-        }), {});
+    it("should render 'General information' Accordion", () => {
+      expect(Accordion).toHaveBeenNthCalledWith(1, expect.objectContaining({
+        id: mockGeneralStaffSlipDetailId,
+        open: accordionDefaultStatus.generalInformation,
+      }), {});
+    });
+
+    it('should render "Template content" Accordion', () => {
+      expect(Accordion).toHaveBeenNthCalledWith(2, expect.objectContaining({
+        id: mocktemplateContentId,
+        open: accordionDefaultStatus.templateContent,
+      }), {});
+    });
+
+    it('should call Metadata component', () => {
+      expect(Metadata).toHaveBeenCalledWith(expect.objectContaining({
+        connect: testStripes.connect,
+        metadata: testDefaultProps.initialValues.metadata,
+      }), {});
+    });
 
-        fireEvent.click(screen.getByTestId(testIds.staffSlipsPreviewButton));
+    it('should display StaffSlipAboutSection', () => {
+      expect(screen.getByText('StaffSlipAboutSection')).toBeVisible();
+    });
 
-        expect(PreviewModal).toHaveBeenLastCalledWith(expect.objectContaining({
-          open: true,
+    describe('handleExpandAll method', () => {
+      it('should render components with default accordions statuses', () => {
+        expect(ExpandAllButton).toHaveBeenLastCalledWith(expect.objectContaining({
+          accordionStatus: accordionDefaultStatus,
         }), {});
 
-        fireEvent.click(screen.getByTestId(testIds.previewModal));
+        expect(Accordion).toHaveBeenLastCalledWith(
+          expect.objectContaining({
+            open: accordionDefaultStatus.generalInformation,
+          }), {}
+        );
+      });
+
+      it('should expand all accordions statuses', () => {
+        fireEvent.click(screen.getByTestId(mockTestIds.expandAllButton));
 
-        expect(PreviewModal).toHaveBeenLastCalledWith(expect.objectContaining({
-          open: false,
+        expect(ExpandAllButton).toHaveBeenLastCalledWith(expect.objectContaining({
+          accordionStatus: {
+            generalInformation: false,
+            templateContent: false,
+          },
         }), {});
+
+        expect(Accordion).toHaveBeenLastCalledWith(
+          expect.objectContaining({
+            open: false,
+          }), {}
+        );
       });
```

Comment:
Could you create 2 separated tests to follow the pattern **"one '_expect_' - one test"**?